### PR TITLE
[#8] signup API error handling - bcrypt async await

### PR DESCRIPTION
**Pull request by jinju2828**
### What (if any) features are you implementing?
-
### What (if anything) did you refactor?
-
### Were there any issues that arose?
-

### Refactor checklist
- [ ] Check variable and parameter names: are they descriptive and unique?
- [ ] Is the code DRY? SRP?
- [ ] Is it easy for others to read your code? (explanatory comments needed?)
- [ ] Is the syntax consistent? (e.g. use of semicolons, blank lines)
- [ ] Are commit messages formatted correctly?

### CODE REVIEW

File: backend/server.js [hunk 1]
```diff
@@ -1,11 +1,91 @@
...
+          userId: userId,
+          userEmail: email,
+          userPassword: cryptedPassword,
+        };
+        console.log({ user });
+        //save it in the json file
+        res.status(201).json({
+          message: "User created successfully!",
+          user: user,
+        });
```
[the remark this answers is not in the document]
Comment (jinju2828) [reply backend/server.js:49]: I thought storing data into user.json file can be done after user created successfully.

File: user.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"name":"John","age":30,"email":"john@example.com"}
```
[the remark this answers is not in the document]
Comment (jinju2828) [reply user.json:1]: It was a test file, I will delete it.

File: backend/server.js [hunk 3]
```diff
@@ -1,11 +1,91 @@
...
 
 app.get("/", (req, res) => {
   res.send("Hello, world!");
 });
 
+//try catch
+app.post("/auth/signup", async (req, res) => {
+  const { email, password, password_confirm } = req.body;
+  function isEmail(email) {
+    let emailRegex = /\S+@\S+\.\S+/;
```
[the remark this answers is not in the document]
Comment (jinju2828) [reply backend/server.js:20]: Working on it, thanks!

**Review by YashsviG — APPROVED** [review]

**Review by susannaopal — APPROVED** [review]
The updates to the code in these files looks great, and readable. Thank you!

**Pull request merged**